### ISSUE TITLE
Add a code-only sample

### ISSUE THREAD

**Issue by Kryptos-FR** [xenko3d/xenko#80]
All samples assume the use of the editor to make a game. We should provide at least one sample (let's call it code template) of a simple scene where everything is done programmatically.

**Comment by tebjan** [comment]
That would be awesome! Xenko is alread quite programmer friendly, so it would also be a good learning source to understand the basic parts of the engine.

**Comment by EssenRoc** [comment]
good job!

**Comment by xen2** [comment]
Code-only sample = sample without assets/xkpkg?

The new Xenko as PackageReference approach should help a lot with that, since project can be created on VS.
Note that few issues such as #58 or #12 might make code-only sample even easier/smoother to create.

**Comment by Kryptos-FR** [comment]
@xen2 Yes that's exactly the idea.

It shows that the editor is optional, and that it is possible to build on top of Xenko's engine without the need to support the whole asset system.

For example use Xenko as a backend to a customized (eventually commercial) engine.

Hopefully that would widen our audience and bring more talents.

**Comment by microdee** [comment]
We managed to make some dirty resizable WinForms app which creates a Game instance and uses Xenko.Graphics to render a black triangle, repo soon™:

https://discourse.vvvv.org/t/link-topics-2-3d-engine/16544/23?u=microdee

**Comment by paralaxsd** [comment]
I can confirm that the sample works.

I had some problems with it though:
1.) `c:\code\xenko-window-from-code-master\packages\Xenko.3.0.0.3\Targets\Xenko.targets(4,5): error : The property $(XenkoPlatform) must be defined by the project`
=> comment line 3 in the project file
2.) The project includes references with absolute hint paths to `C:\Users\m\.nuget\packages` but it doesn't seem to be a problem.

**Comment by rds1983** [comment]
I'd suggest adding xenko project template(s) to the nuget. So one could install it through `dotnet new -i Xenko.Game.Template` and use through `dotnet new xenkogame`

**Comment by microdee** [comment]
yeah also this example needs to be completely rewritten now with the new modular xenko nuget packages. Unfortunately I'm deep in something completely different now so not really going to happen soon from my end :(

**Comment by Kryptos-FR** [comment]
Starting point if someone wants to investigate further to make sure it works for all platforms/configurations: 
[Xnk.zip](https://github.com/xenko3d/xenko/files/2739186/Xnk.zip).

**Comment by tebjan** [comment]
added a repo, the game is currently empty and only opens a black window. would be interesting to add some content in order to make it useful. but here it is for reference:
https://github.com/tebjan/Xenko.CodeOnlyGame

**Comment by xen2** [comment]
Created https://github.com/xen2/Xenko.CodeOnlySample
Just need to update the doc so people can find it (and possibly move it to xenko3d github)

**Comment by tebjan** [comment]
great, made the other repo obsolete and linked the new one...

**Comment by rds1983** [comment]
I've tried to build Xenko.CodeOnlySample from the console using dotnet and it failed:
![image](https://user-images.githubusercontent.com/1057289/51080040-c4c50680-1706-11e9-89e3-3ee6bd92cc6e.png)

Is it a bug? Or did I do something wrong?

**Comment by Kryptos-FR** [comment]
@rds1983 It is not a dotnet application, but .net framework.

**Comment by xen2** [comment]
@Kryptos-FR I have added a dotnet app too
@rds1983 this is fixed, will be part of next Xenko beta. You can build from VS in the meantime

**Comment by rds1983** [comment]
Thanks! The new Xenko release "3.1.0.1-beta01-0396" allows to build Xenko.CodeOnlySample from the command line using "dotnet build".

**Comment by tebjan** [comment]
@Enzojz a xenko game is already a normal .NET solution that you can open with visual studio. just press the visual studio button in the game studio menu.

**Comment by xen2** [comment]
Now the only question is where to host it.
I might move it to xenko3d org, and add it to the doc and/or wiki.

**Comment by CharlesWoodhill** [comment]
is it possible to add a code only example like this? 
is it even possible with xenco?
if there are other ways to achieve something like this, how is it done?

`    
    public class PseucdoCodeExample
    {
        Game simulation = null;
        
        struct UserData
        {
            Xenko.OS.OsWindow splashOsWindow;
        }: userData;

        void EntryPoint()
        {
            // simple use delegates
            simulation.OnInit = OnInit;
            simulation.OnStart = OnStart;
            simulation.OnUpdate = OnUpdate;
            simulation.OnPhysicsUpdate = OnPhysicsUpdate;
            simulation.Run(); //or
            simulation.RunAsync();
        }

        void OnInit()
        {
            // display splash

            //create os window myself or use of xenko helper
            userData.splashOsWindow = new Xenko.OS.Windows.CreateOsWindow(settings); // or
            userData.splashOsWindow = new Xenko.OS.Windows.BindToExistingOsWindow(settings, hWnd);

            // get a texture
            Texture tex = Texture.CreateFromImageFile("file/location/splash.png", opts); // eventuell windows only (asset compiler?)
            Texture tex = Texture.Load("file/location/splash.xTexture?"); // an already compiled? texture (done with gamestudio or batch-created)

            Scene splashScene = new Scene();
            splashScene.Entities.Add(new TexturePanel(..., tex) );

            OsViewPort viewPort = new ViewPort(..., expand.all); // viewport os specific?
            viewPort.RootScene = splashScene;
            userData.splashOsWindow.Components.Add(viewPort);

            userData.splashOsWindow.Show(); // display splash, while doin some work


            hWnd = createOsWindow();
            Xenko.OS.OsWindow mainOsWindow = new Xenko.OS.Windows.BindToExistingOsWindow(settings, hWnd);

            // 2 viewports (for ease just top/bottom :) )
            OsViewPort topViewport = new ViewPortComponent(name: "TopViewPort", ...); // resizeable if window is resized or toggle-fullscreen
            OsViewPort bottomViewport = new ViewPortComponent(name: "BottomViewPort", ...);

            mainOsWindow.Components.Add(topViewport);
            mainOsWindow.Components.Add(bottomViewport);

            // create main windows scene (mostly gui i think) 
            Scene mainViewTopScene = new Scene();
            //....
            topViewport.rootScene = mainViewTopScene;

            // bottom Viewport displays results and stays empty for now

            // setup mvvm or whatever

            userData.splashOsWindow.Free(); // or hide ...
            mainOsWindow.Show();
        }`

**Comment by NicusorN5** [comment]
Since https://github.com/stride3d/stride/issues/1295 is already finished, and the code only approach works fine, should this thread be closed?

**Comment by VaclavElias** [comment]
Maybe, a console like template should be added to our example Stride projects to showcase this? It shouldn't reference the toolkit but use the same-ish code to demo how to do it and anyone could do whatever they want to do with that. Continue or switch to toolkit if desired.

**Comment by NicusorN5** [comment]
What about a toolkit and non-toolkit samples?

**Comment by VaclavElias** [comment]
If we are talking about this yes. Not sure if it should be under samples or templates?

![image](https://github.com/stride3d/stride/assets/4528464/5e9d21e8-b8c1-41ab-ae9c-03c7b350b980)

Non-toolkit sample would be ok, because the code shouldn't change that frequently.

I would recommened a toolkit sample once the toolkit is stable so we don't have to keep updating the template 🤣.

**Comment by IXLLEGACYIXL** [comment]
> If we are talking about this yes. Not sure if it should be under samples or templates?
> 
> ![image](https://private-user-images.githubusercontent.com/4528464/332975072-5e9d21e8-b8c1-41ab-ae9c-03c7b350b980.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0MTYxMzAsIm5iZiI6MTcxNjQxNTgzMCwicGF0aCI6Ii80NTI4NDY0LzMzMjk3NTA3Mi01ZTlkMjFlOC1iOGMxLTQxYWItYWU5Yy0wM2M3YjM1MGI5ODAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MjJUMjIxMDMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzY3ZDlhMGY1OGZjOWZmMjU4Yzc3NzVmMTU1ZTYyZmM3ZTk0ZDdjMjE0ZGY2NWI4ZmIxNDI0MmIxNjE3NmMyMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6YT8UhXf9ehD3C7De4ti7kzRUrG8Wh2qyapV7HnFQWM)
> 
> Non-toolkit sample would be ok, because the code shouldn't change that frequently.
> 
> I would recommened a toolkit sample once the toolkit is stable so we don't have to keep updating the template 🤣.

yes as it may be possible we have to separate code only from the toolkit at some point for example with the skybox problem so the state is currently unknown of the toolkit

**Comment by VaclavElias** [comment]
Also, wasn't there another PR (reversed) for another kind of code-only option?
